backend/infrahub/git/tasks.py

```diff
@@ -195,12 +202,20 @@ async def generate_artifact_definition(branch: str) -> None:
         )
 
 
-@flow(name="artifact-generate")
+@flow(
+    name="artifact-generate",
+    description="Generate a single artifact",
+    flow_run_name="Generate Artifact: {model.artifact_name} for {model.target_name}",
+)
 async def generate_artifact(model: RequestArtifactGenerate) -> None:
-    log.debug("Generating artifact", message=model)
+    await add_branch_tag(branch_name=model.branch_name)
+    # NOTE Not sure if we should use the target_id, the artifact or the artifact definition as the related node
+    await add_related_node_tag(node_id=model.target_id)
```

Comment:
could we do all three or can there only be one?

Reply:
We could do all 3 in prefect but right now the API on Infrahub support only one